Conda troubleshooting (#37)
mantidproject/www: source/installation/condafaq.rst

:orphan:

Conda Troubleshooting
=====================

Before using the troubleshooting below please ensure you are using Mambaforge. Using mamba is preferable over conda as it's faster and better at resolving dependencies.
To install mambaforge please follow `this documentation <https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html>`__

Issues
------

- :ref:`Unable to find mantid or mantidworkbench <unable_to_find>`
- :ref:`Installing on Apple silicon devices <osx_install>`
- :ref:`Unable to resolve dependencies <dependency_fail>`
- :ref:`404 error when trying to install <404_error>`

.. _unable_to_find:

Unable to find `mantid` or `mantidworkbench`
############################################

.. code-block:: sh
	PackagesNotFoundError: The following packages are not available from current channels

> If you get this error it may be because you have not included the mantid channel in your download command. Please ensure you include `-c mantid` when installing.

Reviewer: ```suggestion
If you encounter this error it may be because you have not included the mantid channel in your download command. Please ensure you include `-c mantid` when installing.
```

Example

.. code-block:: sh

   mamba create -n mantid_env -c mantid mantid

> You can also get this error when trying to install on Apple silicon devices. See :ref:`Installing on Apple silicon devices <osx_install>` for help.

Reviewer: ```suggestion
You may also encounter this error when trying to install on Apple silicon devices. See :ref:`Installing on Apple silicon devices <osx_install>` for help.
```

Not sure if this sentence is required because it's right next to the bit that it links to, but it's not a strong opinion.

Author: I guess I'm trying to future proof in case anything gets added inbetween!

.. _osx_install:

Installing on Apple silicon devices
###################################

For macOS we currently only provide x86 packages. These can still be used by those with Apple silicon devices (M1, M2). You will need to have Rosetta installed and precede the setup commands with 'CONDA_SUBDIR=osx-64'.
For example if you want to install mantidworkbench use the following command:

.. code-block:: sh

	CONDA_SUBDIR=osx-64 mamba create -n mantid_env -c mantid mantidworkbench

.. _dependency_fail:

Unable to resolve dependencies
##############################

> If you are having problems with dependencies we recommend installing Mantid into a new clean environment with no other packages. Please do not specify any dependency versions e.g. Python. Any dependencies will be installed automatically of the version required to use Mantid.

Reviewer: ```suggestion
If you are having problems with dependencies we recommend installing Mantid into a new clean environment with no other packages. Please do not specify any dependency versions e.g. Python. All dependencies will be installed automatically with the version required to use Mantid.
```

Reviewer: minor edit

> Problems with resolving dependencies is often caused by installing other packages into the same environment that require different versions of the same dependency. You can check what dependencies recent stable versions of mantid
> available through conda-forge require by running the following

Reviewer: ```suggestion
Problems with resolving dependencies is often caused by installing other packages into the same environment that require different versions of the same dependency. You can view the direct dependencies of recent stable versions of mantid
by running the following command
```
second sentence is confusing for me

.. code-block:: sh

	mamba search -c mantid mantid --info

> To find out the currrent depedencies being used in the nightly replace the channel name with `mantid/label/nightly` __.

Reviewer: ```suggestion
To list the currrent depedencies being used in the nightly replace the channel name with `mantid/label/nightly` __.
```

.. _404_error:

404 error when trying to install
################################

> Mantid is available through `conda-forge`. To access the Mantid channel please check your `.condarc` file. It should contain the following

Reviewer: ```suggestion
Mantid's dependencies are all available through `conda-forge`. To access the this channel please check your `.condarc` file. It should contain the following
```

.. code-block:: sh

	channels:
		- conda-forge

or

.. code-block:: sh

	channels: [conda-forge]

If either of these are in the file then please make sure there are no other channels added for accessing Mantid. For example any of the following will cause a 404 error and should be removed from `.condarc`

.. code-block:: sh

	channels:
		- https://anaconda.org/mantid

> 		- mantid

Reviewer: Question: are you sure that having mantid channel here causes a problem? I've not tried it, but it would seem like it would work.

Author: I got this one from here - https://forum.mantidproject.org/t/404-error-when-trying-to-install-from-conda-on-macos/876/9

Still having problems?
######################

> If the above has not resolved your problem please post to our `community forum <https://forum.mantidproject.org>` or e-mail the tean directly on ``mantid-help@mantidproject.org``.

Reviewer: ```suggestion
If the above has not resolved your problem please post to our `community forum <https://forum.mantidproject.org>` or e-mail the team directly on ``mantid-help@mantidproject.org``.
```
:)
